### PR TITLE
Add tag, common dimensions across compliance queries. Closes #60

### LINKED ISSUE #60: Add common and tag dimensions across compliance queries
**Is your feature request related to a problem? Please describe.**

Common & tag dimensions will allow the end user with options to render compliance output based on tags key, connection name, subscription name & region.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Priyanka-Chatterjee-2000**
### Checklist
- [ ] Issue(s) linked

### CODE REVIEW

File: mod.sp [hunk 1]
```diff
@@ -7,6 +7,57 @@ locals {
...
+  %{~ if contains(var.common_dimensions, "connection_name") }, __QUALIFIER___ctx ->> 'connection_name' as connection_name%{ endif ~}
+  %{~ if contains(var.common_dimensions, "tenant_id") }, __QUALIFIER__tenant_id as tenant_id%{ endif ~}
+  EOQ
+
+  # Local internal variable to build the SQL select clause for tag
+  # dimensions. Do not edit directly.
+  tag_dimensions_qualifier_sql = <<-EOQ
+  %{~ for dim in var.tag_dimensions },  __QUALIFIER__tags ->> '${dim}' as "${replace(dim, "\"", "\"\"")}"%{ endfor ~}
+  EOQ
+
```
Comment (rajlearner17) [inline mod.sp:49]: @khushboo9024 as discussed, we should check the compartment name variable as well.

File: README.md [hunk 2]
```diff
@@ -82,6 +82,31 @@ This mod uses the credentials configured in the [Steampipe OCI plugin](https://h
...
+  ```
+
+- Set an environment variable:
+
+  ```shell
+  SP_VAR_common_dimensions='["connection_name", "region", "tenant"]' steampipe check control.cis_v110_2_1
+  ```
+
+  ```shell
+  SP_VAR_tag_dimensions='[ "Department", "Environment"]' steampipe check control.cis_v110_2_1
```
Comment (misraved) [inline README.md:107]:
```suggestion
  SP_VAR_tag_dimensions='["Department", "Environment"]' steampipe check control.cis_v110_2_1
```

File: docs/index.md [hunk 3]
```diff
@@ -93,6 +93,31 @@ This mod uses the credentials configured in the [Steampipe Oracle Cloud plugin](
...
+  ```
+
+- Set an environment variable:
+
+  ```shell
+  SP_VAR_common_dimensions='["connection_name", "region", "tenant"]' steampipe check control.cis_v110_2_1
+  ```
+
+  ```shell
+  SP_VAR_tag_dimensions='[ "Department", "Environment"]' steampipe check control.cis_v110_2_1
```
Comment (misraved) [inline docs/index.md:118]:
```suggestion
  SP_VAR_tag_dimensions='["Department", "Environment"]' steampipe check control.cis_v110_2_1
```

File: mod.sp [hunk 4]
```diff
@@ -7,6 +7,69 @@ locals {
...
+variable "common_dimensions" {
+  type        = list(string)
+  description = "A list of common dimensions to add to each control."
+  # Define which common dimensions should be added to each control.
+  # - connection_name (_ctx ->> 'connection_name')
+  # - region
+  # - tenant
+  # - tenant_id
+  # - compartment
+  # - compartment_id
```
Comment (misraved) [inline mod.sp:19]: Please sort the list of common dimensions.

**Review by misraved — COMMENTED** [review]
@khushboo9024 please take a look at the minor review comments. Thanks!!

**Review by misraved — APPROVED** [review]

**Pull request merged**